ui/ducks/ens.js

```diff
@@ -0,0 +1,183 @@
+import { createSlice } from '@reduxjs/toolkit';
+import ENS from 'ethjs-ens';
+import log from 'loglevel';
+import networkMap from 'ethereum-ens-network-map';
+import { isConfusing } from 'unicode-confusables';
+import { isHexString } from 'ethereumjs-util';
+
+import { getCurrentChainId } from '../selectors';
+import {
+  CHAIN_ID_TO_NETWORK_ID_MAP,
+  MAINNET_NETWORK_ID,
+} from '../../shared/constants/network';
+import {
+  CONFUSING_ENS_ERROR,
+  ENS_ILLEGAL_CHARACTER,
+  ENS_NOT_FOUND_ON_NETWORK,
+  ENS_NOT_SUPPORTED_ON_NETWORK,
+  ENS_NO_ADDRESS_FOR_NAME,
+  ENS_REGISTRATION_ERROR,
+  ENS_UNKNOWN_ERROR,
+} from '../pages/send/send.constants';
+import { isValidDomainName } from '../helpers/utils/util';
+import { CHAIN_CHANGED } from '../store/actionConstants';
+import {
+  isBurnAddress,
+  isValidHexAddress,
+} from '../../shared/modules/hexstring-utils';
+
+// Local Constants
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
```

Comment:
Can we use `zeroAddress` from `shared/modules/hexstring-utils.js` here?

Reply:
done in latest